### DISCUSSION TITLE
Editing managed files

### DISCUSSION THREAD

**Discussion by bnordgren** [canonical/hpc-team#4] [Support]
What is the intended method to customize managed files like `slurm.conf`? Say I want to partition the cluster or name the main partition something other than slurmd. 

* Do I just edit slurm.conf, confident that juju will not overwrite my changes? 
* Do I use juju somehow to set configuration elements? 
* Is there a command to trigger a sync action for managed files like slurm.conf?

**Comment by jamesbeedy** [comment]
hey @bnordgren, thanks for chiming in!

You can add your own `slurm.conf` parameters via the `slurmctld` charm configuration, `slurm-conf-parameters`.

Given the file `extra.conf`,
```text
SrunProlog=/folder/my_srun_prolog.sh
SrunEpilog=/folder/my_srun_epilog.sh
```
Add these configurations to the `slurm.conf`:
```bash
juju config slurmctld slurm-conf-parameters="$(cat extra.conf)"
```

Regarding the synchronization of the slum.conf file, Juju automatically handles this by reconfiguring and restarting `slurmctld` when changes to the `slum.conf` file are made using the above command.

Hope this helps!

**Reply by jamesbeedy** [reply]
Ahh and the partition name, the `slurm-charms` map the named `slurmd` application charm name to the partition name. So if you want to deploy a cluster with a different partition, you can:

```bash
juju deploy slurmctld
juju deploy slurmd my-special-partition-name
juju relate my-special-partition-name slurmctld
```
